akka-http-core/src/main/scala/akka/http/impl/util/Java6Compat.scala

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2009-2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+
+package akka.http.impl.util
+
+import java.lang.reflect.{ InvocationTargetException, Method }
+import javax.net.ssl.SSLParameters
+
+import scala.util.control.NonFatal
+
+/**
+ * INTERNAL API
+ *
+ * Enables accessing SslParameters even if compiled against Java 6.
+ */
+private[http] object Java6Compat {
```

Comment:
I thought about an `EnhancedSSLParameters`, too, but it seems a little overkill for a one-time use.

`getHostStringJava6Compatible` is used several times, but still I would rather move `getHostStringJava6Compatible` here (for the reasons you mentioned above) than the other way round.